statsd/statsd.go

```diff
@@ -241,6 +245,26 @@ func (s *Server) RunWithCustomSocket(ctx context.Context, sf SocketFactory) erro
 	return ctx.Err()
 }
 
+type handler struct {
+	dispatcher Dispatcher
+	backends   []backendTypes.Backend
+}
+
+func (h *handler) DispatchMetric(ctx context.Context, m *types.Metric) error {
+	return h.dispatcher.DispatchMetric(ctx, m)
+}
+
+func (h *handler) DispatchEvent(ctx context.Context, e *types.Event) error {
+	for _, backend := range h.backends {
+		go func() {
```

Comment:
Ideally we want to have some sort of limit on the number of goroutines we want to spawn concurrently even if they are cheap to avoid OOM/etc in case of event flood. Later :)